protobuf/v1/replica.proto

```diff
@@ -0,0 +1,84 @@
+syntax = "proto3";
+
+import "google/protobuf/struct.proto";
+import "google/protobuf/empty.proto";
+import "common.proto";
+
+package mayastor.v1;
+
+service ReplicaRpc {
+  // Replica related methods.
+  //
+  // Replica allocates space from storage pool.
+  rpc CreateReplica (CreateReplicaRequest) returns (Replica) {}
+  rpc DestroyReplica (DestroyReplicaRequest) returns (google.protobuf.Empty) {}
+  rpc ListReplicas (ListReplicaOptions) returns (ListReplicaResponse) {}
+  rpc ShareReplica (ShareReplicaRequest) returns (ShareReplicaResponse) {}
```

Comment:
Have removed the `StatReplicas` method as we are not sure what this should do. [Ref](https://github.com/openebs/mayastor/blob/develop/mayastor/src/grpc/mayastor_grpc.rs#L533). Will add if the need arises in future for the `Stat` method.

Reply:
Can ShareReplica just return a Replica, since it contains the share_uri?

Reply:
Yes, I think that makes sense, Will update it.

Reply:
Should we have an UnshareReplica rather than using share with Share NONE?

Reply:
It will be similar to the [`Unshare`](https://github.com/openebs/mayastor-api/blob/master/protobuf/v1/bdev.proto#L19) in bdev right?